tox.ini

```diff
@@ -32,14 +32,21 @@ deps = https://github.com/willkg/dennis/archive/aca57f6d.zip
 commands = dennis-cmd lint --errorsonly locale/
 
 [testenv:docker]
-commands = 
-    docker-compose -f docker-compose.yml -f scripts/docker-compose.travis.yml up -d
-    docker exec -i kuma_web_1 make compilejsi18n collectstatic
-    # TODO: mysql needs some time to set up. find a way to eliminate the waiting time
-    sleep 60
-    docker logs kuma_mysql_1
-    docker exec -it kuma_web_1 ./manage.py migrate
-    docker exec -i kuma_web_1 make test
+whitelist_externals =
+    docker-compose
+    sleep
+setenv =
+    COMPOSE_FILE = docker-compose.yml:scripts/docker-compose.travis.yml
+    COMPOSE_PROJECT_NAME = kumatox
+commands =
+    docker-compose build mysql  # Add custom collation to MySQL
+    docker-compose up -d mysql  # Give MySQL time to init DB
+    docker-compose build web    # Rebuild w/ new reqs, Dockerfile-base
+    docker-compose up -d        # Startup remainin services
```

Comment:
I've tried this in PR #4011, and I'm not as happy with the results.

With this version, the ``mysql`` image is built, then run without building, and then then ``web`` image is built, and then everything else runs without a build step.

With ``docker-compeose up --build -d``, the ``mysql`` image is rebuilt, and there are build steps should for ``web``, ``worker``, and ``api``.  The steps are cached and the results are the same.

I prefer this version, and the explicit steps.